ports/mimxrt/mimxrt_sdram.c

```diff
@@ -209,62 +250,38 @@ void mimxrt_sdram_init(void) {
     SEMC_Init(SEMC, &semc_cfg);
 
     #if defined(MIMXRT117x_SERIES)
-
     uint32_t clock_freq = CLOCK_GetRootClockFreq(kCLOCK_Root_Semc);
-
-    semc_sdram_config_t sdram_cfg = {
-        .csxPinMux = kSEMC_MUXCSX0,
-        .address = 0x80000000,
-        .memsize_kbytes = (MICROPY_HW_SDRAM_SIZE >> 10),
-        .portSize = kSEMC_PortSize32Bit, // two 16-bit SDRAMs make up 32-bit portsize
-        .burstLen = kSEMC_Sdram_BurstLen8,
-        .columnAddrBitNum = kSEMC_SdramColunm_9bit,
-        .casLatency = kSEMC_LatencyThree,
-        .tPrecharge2Act_Ns = 15, // tRP 15ns
-        .tAct2ReadWrite_Ns = 15, // tRCD 15ns
-        .tRefreshRecovery_Ns = 70, // Use the maximum of the (Trfc , Txsr).
-        .tWriteRecovery_Ns = 2,  // tWR 2ns
-        .tCkeOff_Ns = 42, // The minimum cycle of SDRAM CLK off state. CKE is off in self refresh at a minimum period tRAS.
-        .tAct2Prechage_Ns = 40, // tRAS 40ns
-        .tSelfRefRecovery_Ns = 70,
-        .tRefresh2Refresh_Ns = 60,
-        .tAct2Act_Ns = 2, // tRC/tRDD 2ns
-        .tPrescalePeriod_Ns = 160 * (1000000000 / clock_freq),
-        .refreshPeriod_nsPerRow = 64 * 1000000 / 8192, // 64ms/8192
-        .refreshUrgThreshold = sdram_cfg.refreshPeriod_nsPerRow,
-        .refreshBurstLen = 1,
-        .delayChain = 2,
-    };
-
     #else
-
     uint32_t clock_freq = CLOCK_GetFreq(kCLOCK_SemcClk);
+    #endif
+
     semc_sdram_config_t sdram_cfg = {
         .csxPinMux = kSEMC_MUXCSX0,
         .address = ((uint32_t)&__sdram_start),
         .memsize_kbytes = (MICROPY_HW_SDRAM_SIZE >> 10),  // Right shift by 10 == division by 1024
-        .portSize = kSEMC_PortSize16Bit,
-        .burstLen = kSEMC_Sdram_BurstLen1,
-        .columnAddrBitNum = kSEMC_SdramColunm_9bit,
-        .casLatency = kSEMC_LatencyThree,
-        .tPrecharge2Act_Ns = 18,  // Trp 18ns
-        .tAct2ReadWrite_Ns = 18,  // Trcd 18ns
-        .tRefreshRecovery_Ns = (60 + 67),
-        .tWriteRecovery_Ns = 12,  // 12ns
-        .tCkeOff_Ns = 42,  // The minimum cycle of SDRAM CLK off state. CKE is off in self refresh at a minimum period tRAS.
-        .tAct2Prechage_Ns = 42,  // Tras 42ns
-        .tSelfRefRecovery_Ns = 67,
-        .tRefresh2Refresh_Ns = 60,
-        .tAct2Act_Ns = 60,
-        .tPrescalePeriod_Ns = 160 * (1000000000 / clock_freq),
+        .portSize = MICROPY_HW_SDRAM_MEM_BUS_WIDTH,
+        .burstLen = MICROPY_HW_SDRAM_BURST_LENGTH,
+        .columnAddrBitNum = MICROPY_HW_SDRAM_COLUMN_BITS_NUM,
+        .casLatency = MICROPY_HW_SDRAM_CAS_LATENCY,
+        .tPrecharge2Act_Ns = MICROPY_HW_SDRAM_TIMING_TRP,
+        .tAct2ReadWrite_Ns = MICROPY_HW_SDRAM_TIMING_TRCD,
+        .tRefreshRecovery_Ns = MICROPY_HW_SDRAM_TIMING_TXSR,
+        .tSelfRefRecovery_Ns = MICROPY_HW_SDRAM_TIMING_TXSR,
```

Comment:
Not clear what tRefreshRecovery_Ns and tSelfRefRecovery_Ns are. Seems like they are TXSR. Not sure why there are two. Previous value of 60 + 67 seems wrong. Not value that high on the datasheet.

So, I think the above changes are correct.

Reply:
Looking at other datasources... this is the refresh to active time. So, seems correct. Same value I guess.